### PR TITLE
Add support for viewing avatar images in full size

### LINKED ISSUE #241: Add support for viewing avatars in full resolution
Nheko unfortunately doesn't allow you to show the full avatar of a user currently, just a slightly enlarged version. Other clients, at least Riot and Element, even allow viewing animated GIFs as animated when viewed in full resolution. Support for at least the former, and perhaps the latter would be nice.

### PR DESCRIPTION
**Pull request by kamathmanu**
- Since GIFs are not really supported in Nheko due to [this bug](https://bugreports.qt.io/browse/QTBUG-30524), this currently only supports static images. 

This should resolve #241

### CODE REVIEW

**Comment by deepbluev7** [comment]
Can you run make lint on this? Otherwise this looks correct! :+1:

File: resources/qml/Avatar.qml [hunk 1]
```diff
@@ -49,6 +51,14 @@ Rectangle {
 
         }
 
+        MouseArea {
+            id: mouseArea
+
+            anchors.fill: parent
+            onClicked: {
+                TimelineManager.openImageOverlay(usrUrl, TimelineManager.timeline.data.id)
+            }
```
Comment (deepbluev7) [inline resources/qml/Avatar.qml:60]:
```suggestion
            onClicked: TimelineManager.openImageOverlay(TimelineManager.timeline.avatarUrl(userid), TimelineManager.timeline.data.id)
```

File: resources/qml/Avatar.qml [hunk 2]
```diff
@@ -9,6 +9,8 @@ Rectangle {
     property alias url: img.source
     property string userid
     property string displayName
+ 
+    property string usrUrl: TimelineManager.timeline.avatarUrl(userid)
```
Comment (deepbluev7) [inline resources/qml/Avatar.qml:13]:
```suggestion
```

File: src/timeline/TimelineViewManager.cpp [hunk 3]
```diff
@@ -339,6 +339,7 @@ TimelineViewManager::toggleCameraView()
 void
 TimelineViewManager::openImageOverlay(QString mxcUrl, QString eventId) const
 {
+        if (mxcUrl.isEmpty() || mxcUrl.isNull()) { return; }
```
Comment (deepbluev7) [inline src/timeline/TimelineViewManager.cpp:342]:
```suggestion
        if (mxcUrl.isEmpty()) { return; }
```

isEmpty returns `true`, if the string is `null`.

grafik

**Comment by kamathmanu** [comment]
Ran make lint and it should be good, thanks!

**Comment by deepbluev7** [comment]
Seems to work perfectly, thanks a lot!

**Pull request merged**